include/GIL/Instructions/AllocaInst.hpp

```diff
@@ -0,0 +1,65 @@
+#ifndef GLU_GIL_INSTRUCTIONS_ALLOCATION_HPP
+#define GLU_GIL_INSTRUCTIONS_ALLOCATION_HPP
+
+#include "AST/ASTContext.hpp"
+#include "AST/ASTNode.hpp"
+#include "InstBase.hpp"
+#include "Type.hpp"
+
+namespace glu::gil {
+
+///
+///  @class AllocaInst
+///  @brief Represents an allocation instruction.
+///
+///  This class is derived from InstBase and represents an allocation
+///  instruction in the GLU GIL (Generic Intermediate Language).
+///
+class AllocaInst : public InstBase {
+    using Context = glu::ast::ASTContext;
+    Context *_context; ///< The AST Context.
+    Type _ptr; ///< The pointer type.
+public:
+    ///
+    /// @brief Constructs an AllocaInst object.
+    ///
+    /// @param type The type of the allocation.
+    /// @param context The AST Context.
+    ///
+    AllocaInst(Type type, Context *context)
+        : InstBase(InstKind::AllocaInstKind)
+        , _context(context)
+        , _ptr(Type(
+              sizeof(void *), alignof(void *), false,
+              _context->getTypesMemoryArena().allocate<glu::types::PointerTy>(
+                  type.getType()
+              )
+          ))
+    {
+    }
+
+    virtual size_t getResultCount() const override { return 1; }
+
+    virtual size_t getOperandCount() const override { return 1; }
+
+    virtual Operand getOperand(size_t index) const override
+    {
+        assert(index == 0 && "Operand index out of range");
+        return Operand(_ptr);
```

Comment:
the operand is the pointee type, not the pointer type (we allocate a value, not a pointer)